examples/OpenTracing.Examples/CommonRequestHandler/HandlerTest.cs

```diff
@@ -0,0 +1,131 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OpenTracing.Mock;
+using OpenTracing.Tag;
+using Xunit;
+
+using static OpenTracing.Examples.TestUtils;
+
+// There is only one instance of 'RequestHandler' per 'Client'. Methods of 'RequestHandler' are
+// executed concurrently in different threads which are reused (common pool). Therefore we cannot
+// use current active span and Activate. So one issue here is setting correct parent span.
+namespace OpenTracing.Examples.CommonRequestHandler
+{
+    public class HandlerTest
+    {
+        private readonly MockTracer _tracer = new MockTracer();
+        private readonly Client _client;
+
+        public HandlerTest()
+        {
+            _client = new Client(new RequestHandler(_tracer));
+        }
+
+        [Fact]
+        public void TwoRequests()
+        {
+            var responseTask = _client.Send("message");
+            var responseTask2 = _client.Send("message2");
+
+            responseTask.Wait(DefaultTimeout);
+            responseTask2.Wait(DefaultTimeout);
+            Assert.Equal("message:response", responseTask.Result);
+            Assert.Equal("message2:response", responseTask2.Result);
+
+            var finished = _tracer.FinishedSpans();
+            Assert.Equal(2, finished.Count);
+
+            Assert.Equal(Tags.SpanKindClient, finished[0].Tags[Tags.SpanKind.Key]);
+            Assert.Equal(Tags.SpanKindClient, finished[1].Tags[Tags.SpanKind.Key]);
+
+            Assert.NotEqual(finished[0].Context.TraceId, finished[1].Context.TraceId);
+            Assert.Equal(0, finished[0].ParentId);
+            Assert.Equal(0, finished[1].ParentId);
+
+            Assert.Null(_tracer.ScopeManager.Active);
+        }
+
+        // active parent is not picked up by child.
+        [Fact]
+        public void ParentNotPickedUp()
+        {
+            using (IScope scope = _tracer.BuildSpan("parent").StartActive(finishSpanOnDispose:true))
+            {
+                var responseTask = _client.Send("no_parent");
```

Comment:
`string response = await _client.Send("no_parent");` (and async method signature)